app.py

```diff
@@ -2,56 +2,84 @@
-    asg_stack = FoldingAsgStack(app, "folding-asg",
-                                region=config['aws']['region'],
-                                vpc=vpc_stack.vpc,
-                                ec2_instance_type=config['aws']['ec2_instance_type'],
-                                ami_id=config['aws']['ami_id'],
-                                ssh_key=config['aws']['ssh_key'],
-                                max_spot_price=config['aws']['max_spot_price'],
-                                ssh_allow_ip_range=config['aws']['ssh_allow_ip_range'],
-                                asg_size=config['aws']['asg_size'],
-                                force_spot_price=force_spot_price,
-                                env=core.Environment(region=config['aws']['region']))
-    app.synth()
+  config = { stack_name : {
+                            **defaults[stack_name], 
+                            **user_config[stack_name],
+                          }
+          }
+  return config[stack_name]
+
+def cdk_init(stack_name: str, force_spot_price: bool):
+  config = configure(stack_name)
+  tags = config.get('tags')
+  vpc_stack = VpcStack(app, 
+                      f"{stack_name}-vpc",
+                      cidr=config['cidr'],
+                      env=core.Environment(region=config['region'],
+                      )
+                    )
+  asg_stack = AsgStack(app, 
+                    f"{stack_name}-asg", 
+                    stack_name=stack_name,
+                    region=config['region'], 
+                    vpc=vpc_stack.vpc, 
+                    ec2_instance_type=config['ec2_instance_type'], 
+                    ami_id=config['ami_id'],
+                    ssh_key=config['ssh_key'], 
+                    max_spot_price=config['max_spot_price'],
+                    ssh_allow_ip_range=config['ssh_allow_ip_range'],
+                    asg_size=config['asg_size'],
+                    force_spot_price=force_spot_price,
+                    env=core.Environment(region=config['region'])
+                  )
+  
+  for tag in tags:
+    core.Tag.add(asg_stack, tag.get('name'), tag.get('value'))  
+    core.Tag.add(vpc_stack, tag.get('name'), tag.get('value'))
+
+  app.synth()
 
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser(description="Manage a Folding@Home EC2 stack")
-    parser.add_argument('--force', action="store_true",
-                        help="Use the spot price specified in config file, do not query AWS for current spot prices")
-    args = parser.parse_args()
-    cdk_init(configure(), args['force'])
+  app = core.App()
+  stack_name = app.node.try_get_context("stack_name")
+  
+  if stack_name is None:
+    print("Please pass a stack name using -c <stack name>")
+    sys.exit(1)
+  
+  force_spot_price = app.node.try_get_context("force_spot")
+  if force_spot_price:
+    print("Force Spot price flag is set, will auto fetch spot price")
+    print(force_spot_price)
```

Comment:
Was this a debug print, that we can skip in the actual PR?
```suggestion
```

Reply:
woops. debug.